### PR TITLE
gh: migrate to using github issue forms

### PR DESCRIPTION
**Pull request by abn**
Ref: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        Please attach output from `poetry --version`
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Poetry Configuration
+      description: |
+        Please attach output from `poetry config --list`
+      render: 'bash session'
```
Comment (neersighted) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:71]: Should this be applied to `--version` as well?

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      label: Python Sysconfig
+      description: |
+        Please attach output from `python -m sysconfig`
+      render: 'bash session'
+    validations:
+      required: false
+
+  - type: textarea
+    attributes:
+      label: Example PyProject TOML
```
Comment (neersighted) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:86]:
```suggestion
      label: Example pyproject.toml
```

File: .github/ISSUE_TEMPLATE/---feature-request.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        Also, ensure that the issue is not already fixed in the [development documentation](https://python-poetry.org/docs/main/).
+      render: "Markdown"
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Impact
+      description: |
+        Please describe the motivation for this issue. Describe, as best you can, ow this improves or impacts the users
```
Comment (neersighted) [inline .github/ISSUE_TEMPLATE/---feature-request.yml:43]:
```suggestion
        Please describe the motivation for this issue. Describe, as best you can, how this improves or impacts the users
```

File: .github/ISSUE_TEMPLATE/---feature-request.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        of Poetry and why this is important.
+      render: "Markdown"
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Workarounds
+      description: |
+        Is there a mitigation or workaround that allows users to achieve the same functionality today?
```
Comment (neersighted) [inline .github/ISSUE_TEMPLATE/---feature-request.yml:53]:
```suggestion
        Is there a mitigation, workaround, or addon that allows users to achieve the same functionality today?
```

File: .github/ISSUE_TEMPLATE/---bug-report.yml (hunk 1 above)
Comment (abn) [reply .github/ISSUE_TEMPLATE/---bug-report.yml:71]: Cannot because it is of type `input`, this is `textarea`.

**Review by neersighted — APPROVED** [review]

**Pull request merged**